pkg/virt-launcher/virtwrap/live-migration-source.go

```diff
@@ -605,23 +655,30 @@ func (m *migrationMonitor) startMonitor() {
 		time.Sleep(400 * time.Millisecond)
 
 		err := m.hasMigrationErr()
-		if err != nil {
-			logger.Reason(err).Error("Live migration failed")
+		if err != nil && m.migrationFailedWithError == nil {
+			logger.Reason(err).Error("Recevied a live migration error. Will check the latest migration status.")
+			m.migrationFailedWithError = err
+			continue
+		} else if m.migrationFailedWithError != nil {
+			logger.Info("Didn't manage to get a job status. Post the received error and finilize.")
+			logger.Reason(m.migrationFailedWithError).Error("Live migration failed")
 			var abortStatus v1.MigrationAbortStatus
-			if strings.Contains(err.Error(), "canceled by client") {
+			if strings.Contains(m.migrationFailedWithError.Error(), "canceled by client") {
 				abortStatus = v1.MigrationAbortSucceeded
 			}
-			m.l.setMigrationResult(vmi, true, fmt.Sprintf("Live migration failed %v", err), abortStatus)
+			m.l.setMigrationResult(vmi, true, fmt.Sprintf("Live migration failed %v", m.migrationFailedWithError), abortStatus)
 			return
 		}
```

Comment:
because of the `continue` statement, I think these two `if` and `else if` statements are redundant.

For example, lets say the first if statement is hit, we'll set `m.migrationFailedWithError` then execute the `continue` which takes us to the top of the for loop. the second iteration of will then always hit the `else if` statement here because `m.migrationFailedWithError != nil`

Functionally, this logic appears to be the same as the previous code, but with a sleep. For example, I think the code below would be exactly the same as what you have here at the moment. 

```
if err != nil {
                        logger.Reason(err).Error("Live migration failed")

                        time.Sleep(400 * time.Millisecond)

                        var abortStatus v1.MigrationAbortStatus
                        if strings.Contains(err.Error(), "canceled by client") {
                                abortStatus = v1.MigrationAbortSucceeded
                        }
                        m.l.setMigrationResult(vmi, true, fmt.Sprintf("Live migration failed %v", err), abortStatus)
                        return

}
```